pkg/l2announcer/l2announcer.go

```diff
@@ -311,8 +311,19 @@ func (l2a *L2Announcer) processPolicyEvent(ctx context.Context, event resource.E
 }
 
 func (l2a *L2Announcer) upsertSvc(svc *slim_corev1.Service) error {
-	// Ignore services managed by an unsupported load balancer class.
 	key := serviceKey(svc)
+
+	// Ignore services if there is no noExternal or LB IP assigned.
+	noExternal := svc.Spec.ExternalIPs == nil
+	noLB := true
+	for _, v := range svc.Status.LoadBalancer.Ingress {
+		noLB = noLB && v.IP == ""
+	}
```

Comment:
This reads odd. I suggest writing it like so which is a more familiar pattern
```suggestion
	noLB := true
	for _, v := range svc.Status.LoadBalancer.Ingress {
		if v.IP != "" {
		    noLB = false
		    break
		}
	}
```

Reply:
The rest of the changes look good to me!

Reply:
Thanks, changed.